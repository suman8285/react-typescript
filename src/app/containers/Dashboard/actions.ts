import {
  USER_BALANCE,
  USER_BALANCE_ERROR,
  USER_BALANCE_SUCCESS,
} from './action_types';
import { IUserBalance, IDashboardState } from './types';

interface IAction {
  type: string;
  payload?: string | number;
}

export const getBalance = (): IAction => {
  return {
    type: USER_BALANCE,
  };
};

export const getBalanceSuccess = (payload: number): IAction => {
  return {
    type: USER_BALANCE_SUCCESS,
    payload,
  };
};

export const getBalanceError = (payload: string): IAction => {
  return {
    type: USER_BALANCE_ERROR,
    payload,
  };
};
