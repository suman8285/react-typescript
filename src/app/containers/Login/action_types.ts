export const AUTH_INITIALIZE = 'AUTH_INITIALIZE';
export const AUTH_LOGGING_IN = 'AUTH_LOGGING_IN';
export const AUTH_LOGGED_IN = 'AUTH_LOGGED_IN';
export const AUTH_ERROR = 'AUTH_ERROR';
export const AUTH_LOGOUT = 'AUTH_LOGOUT';
