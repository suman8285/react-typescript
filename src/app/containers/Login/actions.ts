import {
  AUTH_INITIALIZE,
  AUTH_LOGGING_IN,
  AUTH_LOGGED_IN,
  AUTH_ERROR,
  AUTH_LOGOUT,
} from './action_types';
import { IAuthCredentials } from './types';

interface IAction {
  type: string;
  payload?: string | IToken | IAuthCredentials;
}

interface IToken {
  [key: string]: string;
}

export const authInitialize = (payload: IToken): IAction => {
  return {
    type: AUTH_INITIALIZE,
    payload,
  };
};

export const authLoggingIn = (payload: IAuthCredentials): IAction => {
  return {
    type: AUTH_LOGGING_IN,
    payload,
  };
};

export const authLoggedIn = (payload: IToken): IAction => {
  return {
    type: AUTH_LOGGED_IN,
    payload,
  };
};

export const authError = (payload: string): IAction => {
  console.log('in auth error login');

  return {
    type: AUTH_ERROR,
    payload,
  };
};

export const authLogout = (): IAction => {
  return {
    type: AUTH_LOGOUT,
  };
};
