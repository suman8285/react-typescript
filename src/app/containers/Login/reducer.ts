import { TOKEN_KEY, NAME } from 'app/components/common/Constants';
import {
  AUTH_INITIALIZE,
  AUTH_LOGGING_IN,
  AUTH_LOGGED_IN,
  AUTH_ERROR,
  AUTH_LOGOUT,
} from './action_types';
import { IAuthState } from './types';

//The initial state of the Auth container
export const initialState: IAuthState = {
  isAuthenticated: false,
  token: '',
  isLoggingIn: false,
  loginError: '',
  initialized: false,
  username: '',
};

export const reducer = (state = initialState, action): IAuthState => {
  switch (action.type) {
    case AUTH_INITIALIZE:
      const { token } = action.payload;
      if (token) {
        return {
          ...state,
          initialized: true,
          isLoggingIn: false,
          isAuthenticated: true,
          token,
          loginError: '',
        };
      } else {
        return {
          ...state,
          initialized: true,
          isLoggingIn: false,
          isAuthenticated: false,
          loginError: '',
        };
      }
    case AUTH_LOGGING_IN:
      return {
        ...state,
        username: action.payload.username,
        isLoggingIn: true,
        loginError: '',
      };
    case AUTH_LOGGED_IN:
      window.localStorage.setItem(TOKEN_KEY, action.payload.token);
      return {
        ...state,
        initialized: true,
        isLoggingIn: false,
        isAuthenticated: true,
        token: action.payload.token,
        loginError: '',
        username: action.payload.username,
      };
    case AUTH_ERROR:
      return {
        ...state,
        initialized: true,
        isAuthenticated: false,
        isLoggingIn: false,
        loginError: action.payload,
        token: '',
        username: '',
      };
    case AUTH_LOGOUT:
      window.localStorage.removeItem(TOKEN_KEY);
      return {
        ...state,
        initialized: true,
        isAuthenticated: false,
        token: '',
      };
    default:
      return state;
  }
};

export const key = 'auth';
