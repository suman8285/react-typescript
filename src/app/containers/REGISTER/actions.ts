import {
  USER_REGISTRATION,
  USER_REGISTRATION_ERROR,
  USER_REGISTRATION_SUCCESS,
} from './action_types';
import { IRegisterUser } from './types';

interface IAction {
  type: string;
  payload?: string | IRegisterUser;
}

export const userRegistring = (payload: IRegisterUser): IAction => {
  console.log('In this');

  return {
    type: USER_REGISTRATION,
    payload,
  };
};

export const userRegistrationSuccess = (): IAction => {
  console.log('in action2');

  return {
    type: USER_REGISTRATION_SUCCESS,
  };
};

export const userRegistrationError = (payload: string): IAction => {
  console.log('in action2', payload);

  return {
    type: USER_REGISTRATION_ERROR,
    payload,
  };
};
