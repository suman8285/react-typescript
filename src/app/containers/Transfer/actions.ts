import {
  USER_TRANSFER,
  USER_TRANSFER_ERROR,
  USER_TRANSFER_SUCCESS,
  RESET,
} from './action_types';
import { ITransferUser } from './types';

interface IAction {
  type: string;
  payload?: string | ITransferUser;
}

export const userTransfering = (payload: ITransferUser): IAction => {
  return {
    type: USER_TRANSFER,
    payload,
  };
};

export const userTransferSuccess = (payload: string): IAction => {
  return {
    type: USER_TRANSFER_SUCCESS,
    payload,
  };
};

export const userTransferError = (payload: string): IAction => {
  return {
    type: USER_TRANSFER_ERROR,
    payload,
  };
};

export const reset = (): IAction => {
  return {
    type: RESET,
  };
};
