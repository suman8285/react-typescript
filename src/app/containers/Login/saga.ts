import { put, takeLatest, call } from 'redux-saga/effects';
import { PayloadAction } from '@reduxjs/toolkit';
import { request } from 'utils/request';
import { LoginEndpoint } from 'app/components/common/Constants';
import { IAuthCredentials, ILoginUser } from './types';
import { AUTH_LOGGING_IN } from './action_types';
import { authLoggedIn, authError } from './actions';

interface IResponse {
  data: ILoginUser;
}

export function* getToken(action: PayloadAction<IAuthCredentials>): Generator {
  try {
    const { username, password } = action.payload;
    const { data } = (yield call(request, {
      url: LoginEndpoint,
      method: 'post',
      data: {
        userId: username,
        password,
      },
    })) as IResponse;

    yield put(authLoggedIn({ token: data.token, username: data.username }));
  } catch (err) {
    const errorMessage =
      err.response.status === 404
        ? 'User Not Found'
        : err.response.data.message;
    yield put(authError(errorMessage));
  }
}

/**
 * Root saga manages watcher lifecycle
 */
export function* authSaga(): Generator {
  // Watches for loadRepos actions and calls getRepos when one comes in.
  // By using `takeLatest` only the result of the latest API call is applied.
  // It returns task descriptor (just like fork) so we can continue execution
  // It will be cancelled automatically on component unmount
  yield takeLatest(AUTH_LOGGING_IN, getToken);
}
