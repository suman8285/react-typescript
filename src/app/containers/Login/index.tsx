import React, { useState, useEffect } from 'react';
import { Helmet } from 'react-helmet';
import { useHistory, Redirect } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components/macro';
import {
  Row as BaseRow,
  Container as BaseContainer,
} from 'react-awesome-styled-grid';
import { useInjectReducer, useInjectSaga } from 'utils/redux-injectors';
import {
  HomeURL,
  RegisterURL,
  TOKEN_KEY,
  DashboardUrl,
} from 'app/components/common/Constants';
import { IRootState } from 'types';
import { key, reducer } from './reducer';
import { authSaga } from './saga';
import { authLoggingIn, authInitialize } from './actions';
import { IAuthState } from './types';
import { media } from 'styles/media';

const Body = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  height: 100vh;
  background-image: linear-gradient(258deg, #93dc6d 5%, #1f9cc3);
  background-repeat: no-repeat;
  background-size: cover;
`;

const Container = styled(BaseContainer)`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  height: 50%;
  color: white;
  padding-left: 0px;
  padding-right: 0px;
  ${media.small`
    width: 100%;
  `}
`;

const Heading = styled.p`
  font-size: 52px;
  font-weight: 100;
  color: ${(props): string => props.theme.black};
  margin: 0;
  padding-top: 5px;
  padding-bottom: 10px;
  ${media.small`
    width: 100%;
    font-size: 32px;
  `}
`;

const SubHeading = styled.p`
  width: 60%
  font-size: 20px;
  font-weight: 100;
  flex-direction: column;
  color: ${(props): string => props.theme.black};
  text-align: center;
  margin: 0;
  padding-top: 10px;
  padding-bottom: 10px;
  ${media.small`
    width: 100%;
    font-size: 15px;
  `}
`;

const Row = styled(BaseRow)`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: 16px;
  background: transparent;
  color: ${(props): string => props.theme.black};
  padding: 20px 0;
`;

const Input = styled.input`
  width: 50%;
  height: 60px;
  padding: 0 20px;
  border-radius: 30px;
  font-size: 14px;
  color: ${(props): string => props.theme.black};
  border: 1px solid ${(props): string => props.theme.darkGray};
  margin-bottom: 10px;
  &:focus-within {
    box-shadow: 0 0 15px ${(props): string => props.theme.gray};
  }
  ${media.small`
    width: 100%;
  `}
`;

const Button = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50%;
  border: 0;
  background-color: green;
  border: 1px solid transparent;
  color: white;
  padding: 20px 0;
  border-radius: 30px;
  font-size: 14px;
  :hover {
    cursor: pointer;
    background: green;
    border: 1px solid green;
  }
  :focus {
    background: green;
    border: 1px solid green;
  }
  ${media.small`
    width: 100%;
  `}
`;

const LoginPage = styled(BaseContainer)`
  width: 960px;
  height: 580px;
  background: #fff;
  flex-direction: column;
  border-radius: 16px;
  text-align: center;
  display: flex;
  justify-content: space-between;
  padding: 80px 90px 25px 95px;
`;

const TopBlackHeader = styled.div`
  height: 64px;
  padding: 0px;
  top: 0;
  background-color: black;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 1.2rem;
`;

const White = styled.div`
  color: white;
`;

const ButtonRegister = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50%;
  border: 0;
  background-color: blue;
  border: 1px solid transparent;
  color: white;
  padding: 20px 0;
  border-radius: 30px;
  font-size: 14px;
  :hover {
    cursor: pointer;
    background: blue;
    border: 1px solid green;
  }
  :focus {
    background: green;
    border: 1px solid green;
  }
  ${media.small`
    width: 100%;
  `}
`;

interface ILoginProps {
  storage: Storage;
}

export function Login({ storage }: ILoginProps): React.ReactElement {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  useInjectReducer({ key, reducer });
  useInjectSaga({ key, saga: authSaga });

  const history = useHistory();
  const dispatch = useDispatch();

  const { isAuthenticated, token, loginError } = useSelector(
    (state: IRootState): IAuthState => state.auth,
  );

  const handleUsername = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setUsername(e.currentTarget.value);
  };

  const handlePassword = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setPassword(e.currentTarget.value);
  };

  const doLogin = (): void => {
    dispatch(authLoggingIn({ username, password }));
  };

  const doRegister = (): void => {
    console.log('history');
    history.push(RegisterURL);
  };

  const authenticate = (): void => {
    if (!isAuthenticated) {
      dispatch(authInitialize({ token: storage.getItem(TOKEN_KEY) || '' }));
    } else {
      if (storage && token) {
        history.push(DashboardUrl);
      }
    }
  };

  useEffect(authenticate, [isAuthenticated, loginError]);

  return (
    <>
      <Helmet>
        <title>Login Page</title>
        <meta
          name="description"
          content="A React Boilerplate application Login page"
        />
      </Helmet>

      <TopBlackHeader>
        <White>
          Cryptocurrency <b>Wallet</b>
        </White>
      </TopBlackHeader>
      <Body>
        <Container>
          <LoginPage>
            <Heading>Log In</Heading>
            <SubHeading>Login to your Diem Wallet account.</SubHeading>
            <Row>
              <Input
                value={username}
                id="username"
                type="text"
                placeholder="username"
                onChange={handleUsername}
              />
              <Input
                value={password}
                id="password"
                type="password"
                placeholder="password"
                onChange={handlePassword}
              />
              <br />
              <Button onClick={doLogin}>LOGIN</Button>
              <SubHeading>Donâ€™t have an account yet?</SubHeading>
              <ButtonRegister onClick={doRegister}>SIGN-UP</ButtonRegister>
            </Row>
          </LoginPage>
        </Container>
      </Body>
    </>
  );
}
