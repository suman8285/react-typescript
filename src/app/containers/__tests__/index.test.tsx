import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Store } from '@reduxjs/toolkit';
import { fireEvent, screen, render, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureAppStore } from 'store/configureStore';
import { authInitialize, authLoggedIn, authError } from '../Login/actions';
import { TOKEN_KEY } from '../../components/common/Constants';
import App from '../index';

interface ILocation {
  state: {
    [key: string]: string;
  };
}

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as object),
  useLocation: (): ILocation => ({
    state: {
      from: '',
    },
  }),
}));

const store: Store = configureAppStore();

describe('Container', () => {
  it('Matching the snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <App storage={window.localStorage} />
        </BrowserRouter>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Matching the snapshot when logging error', () => {
    window.localStorage.removeItem(TOKEN_KEY);
    store.dispatch(authError('User Not Found'));
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <App storage={window.localStorage} />
        </BrowserRouter>
      </Provider>,
    );

    const redirect = screen.queryByRole('Redirect');
    expect(redirect).toBeDefined();

    expect(asFragment()).toMatchSnapshot();
  });

  it('Matching the snapshot when logged in', () => {
    const token =
      'hdgjshjhfldijalkvndjhvali.dbhkavauheiodaji.dbvajhvfaiuguqego';
    store.dispatch(authLoggedIn({ token }));
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <App storage={window.localStorage} />
        </BrowserRouter>
      </Provider>,
    );

    const element = screen.queryByRole('div');
    expect(element).toBeDefined();

    expect(asFragment()).toMatchSnapshot();
  });

  it('Matching the snapshot when loading', () => {
    const token = '';
    store.dispatch(authInitialize({ token }));
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <App storage={window.localStorage} />
        </BrowserRouter>
      </Provider>,
    );

    const element = screen.queryByRole('LoadingView');
    expect(element).toBeDefined();

    expect(asFragment()).toMatchSnapshot();
  });
});
