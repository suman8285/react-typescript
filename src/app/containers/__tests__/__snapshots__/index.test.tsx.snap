// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Matching the snapshot 1`] = `<DocumentFragment />`;

exports[`Container Matching the snapshot when loading 1`] = `
<DocumentFragment>
  .c0 {
  height: 100%;
  width: 100%;
}

.c2 {
  margin-right: auto;
  margin-left: auto;
  max-width: 100%;
  box-sizing: border-box;
}

.c5 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100vh;
  background-image: linear-gradient(258deg,#93dc6d 5%,#1f9cc3);
  background-repeat: no-repeat;
  background-size: cover;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 50%;
  color: white;
  padding-left: 0px;
  padding-right: 0px;
}

.c4 {
  font-size: 52px;
  font-weight: 100;
  text-align: center;
  margin: 0;
  padding-top: 50px;
  padding-bottom: 10px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  background: transparent;
  padding: 20px 0;
}

.c7 {
  border: 0;
  background-color: green;
  border: 1px solid transparent;
  color: white;
  padding: 26px 0;
  text-align: center;
  font-size: 14px;
  width: 250px;
}

.c7:hover {
  cursor: pointer;
  background: green;
  border: 1px solid green;
}

.c7:focus {
  background: green;
  border: 1px solid green;
}

@media only screen and (min-width:1rem) {
  .c2 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media only screen and (min-width:48rem) {
  .c2 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media only screen and (min-width:64rem) {
  .c2 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width:90rem) {
  .c2 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width:120rem) {
  .c2 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width:1rem) {
  .c2 {
    width: 100%;
  }
}

@media only screen and (min-width:48rem) {
  .c2 {
    width: 100%;
  }
}

@media only screen and (min-width:64rem) {
  .c2 {
    width: 100%;
  }
}

@media only screen and (min-width:90rem) {
  .c2 {
    width: 90rem;
  }
}

@media only screen and (min-width:120rem) {
  .c2 {
    width: 90rem;
  }
}

@media only screen and (min-width:1rem) {
  .c5 {
    margin-left: -0.5rem;
    margin-right: -0.5rem;
  }
}

@media only screen and (min-width:48rem) {
  .c5 {
    margin-left: -0.5rem;
    margin-right: -0.5rem;
  }
}

@media only screen and (min-width:64rem) {
  .c5 {
    margin-left: -0.75rem;
    margin-right: -0.75rem;
  }
}

@media only screen and (min-width:90rem) {
  .c5 {
    margin-left: -0.75rem;
    margin-right: -0.75rem;
  }
}

@media only screen and (min-width:120rem) {
  .c5 {
    margin-left: -0.75rem;
    margin-right: -0.75rem;
  }
}

<div
    class="containers__AppContainer-h6s92r-0"
  >
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2 c3"
        >
          <p
            class="c4"
          >
            Welcome Home
          </p>
          <div
            class="c5 c6"
          >
            <button
              class="c7"
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Container Matching the snapshot when logged in 1`] = `
<DocumentFragment>
  .c0 {
  height: 100%;
  width: 100%;
}

<div
    class="containers__AppContainer-h6s92r-0"
  >
    <div
      class="c0"
    >
      <div>
        Loading...
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Container Matching the snapshot when logging error 1`] = `<DocumentFragment />`;
